core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -848,39 +849,44 @@ abstract class RDD[T: ClassTag](
    * of elements in each partition.
    */
   def zipPartitions[B: ClassTag, V: ClassTag]
-      (rdd2: RDD[B], preservesPartitioning: Boolean)
+      (rdd2: RDD[B], preservesPartitioning: Boolean, preferredLocationRDD: RDD[_])
```

Comment:
Make preferredLocationRDD and Option.
Also can you please add some documentation about it ?
Like what happens when it is specified, what happens when it is not ?